### ISSUE TITLE
Heatmap only supports equal row and column

### ISSUE THREAD

**Issue by zxdawn** [lukelbd/proplot#136]
### Description

`heatmap` only works for equal row and column.

### Steps to reproduce
```python
import proplot as plot
import pandas as pd

data = pd.DataFrame(index=list('abcdefghij'), columns=list('lmn'))
data = data.fillna(0)
print(data.values.shape)

# Covariance matrix plot
f, ax = plot.subplots(axwidth=4.5)
m = ax.heatmap(
    data, cmap='ColdHot', vmin=-1, vmax=1, N=100,
    lw=0.5, edgecolor='k', labels=True, labels_kw={'weight': 'bold'},
    clip_on=False, # turn off clipping so box edges are not cut in half
)
ax.format(
    suptitle='Heatmap demo', title='Table of correlation coefficients', alpha=0, linewidth=0,
    xloc='top', yloc='right', yreverse=True, ticklabelweight='bold',
    ytickmajorpad=4,  # the ytick.major.pad rc setting; adds extra space
)
```

**Expected behavior (generated by seaborn)**:
```
import seaborn as sns
import pandas as pd

data = pd.DataFrame(index=list('abcdefghij'), columns=list('lmn'))
data = data.fillna(0)
sns.set()
ax = sns.heatmap(data, cmap="YlGn")
```

![image](https://user-images.githubusercontent.com/30388627/77870493-245bf300-7274-11ea-8f8c-796372596d98.png)

**Actual behavior**:
```
  File "E:\miniconda3\envs\satpy\lib\site-packages\proplot\subplots.py", line 217, in _iterator
    ret.append(func(*args, **kwargs))
  File "E:\miniconda3\envs\satpy\lib\site-packages\proplot\axes.py", line 1275, in heatmap
    obj = self.pcolormesh(*args, **kwargs)
  File "E:\miniconda3\envs\satpy\lib\site-packages\proplot\wrappers.py", line 3083, in _wrapper
    return driver(self, func, *args, **kwargs)
  File "E:\miniconda3\envs\satpy\lib\site-packages\proplot\wrappers.py", line 551, in standardize_2d
    f'Input shapes x {x.shape} and y {y.shape} must match '
ValueError: Input shapes x (10,) and y (3,) must match Z centers (10, 3) or Z borders (11, 4).
```

### Proplot version
0.5.0

**Comment by zxdawn** [comment]
@bradyrx Thanks! It works now ;)

Because I want `Square`, I set the aspect based on the shape of data:
```
f, ax = plot.subplots(aspect=data.shape[::-1])
```

**Comment by bradyrx** [comment]
That's a nice idea. Glad this worked!

**Comment by lukelbd** [comment]
Thanks @zxdawn for the report and @bradyrx for the help! This is a proplot-specific bug and should definitely be fixed, will keep this open until then.

**Comment by lukelbd** [comment]
This is fixed by #159. Note you can also make the grid boxes square by passing `aspect='equal'` to `format()` (which calls [`Axes.set_aspect()`](https://matplotlib.org/3.2.1/api/_as_gen/matplotlib.axes.Axes.set_aspect.html)) although #159 also makes "equal" aspect ratio the default behavior for heatmaps, just like it is the default for `imshow` plots.

**Comment by zxdawn** [comment]
@lukelbd Thanks. Here's the result based on the newest repo:
![heatmap](https://user-images.githubusercontent.com/30388627/81494309-744dc100-92da-11ea-951a-b87219ef1cde.png)

**Code**:
```
import proplot as plot
import pandas as pd
import numpy as np

data = pd.DataFrame(index=list('abcde'), columns=list('lmn'))
data[:] = np.random.rand(5, 3)

f, ax = plot.subplots(axwidth=4.5)
m = ax.heatmap(
    data, cmap='ColdHot', vmin=-1, vmax=1, N=100,
    lw=0.5, edgecolor='k', labels=True, labels_kw={'weight': 'bold'},
    clip_on=False, # turn off clipping so box edges are not cut in half
)
ax.format(
    suptitle='Heatmap demo', title='Table of correlation coefficients', alpha=0, linewidth=0,
    xloc='top', yloc='right', yreverse=True, ticklabelweight='bold',
    ytickmajorpad=4,  # the ytick.major.pad rc setting; adds extra space
)
```
